tiledb/sm/serialization/array.cc

```diff
@@ -358,6 +365,257 @@ Status array_open_from_capnp(
+        break;
+      }
+      default: {
+        throw ArraySerializationStatusException(
+            "[fragments_timestamps_serialize] Unknown serialization type "
+            "passed");
+      }
+    }
+
+  } catch (kj::Exception& e) {
+    throw ArraySerializationStatusException(
+        "[fragments_timestamps_serialize] kj::Exception: " +
+        std::string(e.getDescription().cStr()));
+  } catch (std::exception& e) {
+    throw ArraySerializationStatusException(
+        "[fragments_timestamps_serialize] exception " + std::string(e.what()));
+  }
+}
+
+void fragments_timestamps_deserialize(
+    uint64_t& start_timestamp,
+    uint64_t& end_timestamp,
+    SerializationType serialize_type,
+    const Buffer& serialized_buffer) {
+  try {
+    switch (serialize_type) {
+      case SerializationType::JSON: {
+        ::capnp::JsonCodec json;
+        ::capnp::MallocMessageBuilder message_builder;
+        auto builder =
+            message_builder.initRoot<capnp::ArrayFragmentsTimestamps>();
+        json.decode(
+            kj::StringPtr(static_cast<const char*>(serialized_buffer.data())),
+            builder);
+        auto reader = builder.asReader();
+        // Deserialize
+        fragments_timestamps_from_capnp(reader, start_timestamp, end_timestamp);
+        break;
+      }
+      case SerializationType::CAPNP: {
+        const auto mBytes =
+            reinterpret_cast<const kj::byte*>(serialized_buffer.data());
+        ::capnp::FlatArrayMessageReader msg_reader(kj::arrayPtr(
+            reinterpret_cast<const ::capnp::word*>(mBytes),
+            serialized_buffer.size() / sizeof(::capnp::word)));
+        auto reader = msg_reader.getRoot<capnp::ArrayFragmentsTimestamps>();
+        // Deserialize
+        fragments_timestamps_from_capnp(reader, start_timestamp, end_timestamp);
+        break;
+      }
+      default: {
+        throw ArraySerializationStatusException(
+            "[fragments_timestamps_deserialize] "
+            "Unknown serialization type passed");
+      }
+    }
+  } catch (kj::Exception& e) {
+    throw ArraySerializationStatusException(
+        "[fragments_timestamps_deserialize] kj::Exception: " +
+        std::string(e.getDescription().cStr()));
+  } catch (std::exception& e) {
+    throw ArraySerializationStatusException(
+        "[fragments_timestamps_deserialize] exception " +
+        std::string(e.what()));
+  }
+}
+
+void fragments_list_to_capnp(
+    const std::vector<URI>& fragments,
+    capnp::ArrayFragmentsList::Builder* array_fragments_list_builder) {
+  auto entries_builder =
+      array_fragments_list_builder->initEntries(fragments.size());
+  for (size_t i = 0; i < fragments.size(); i++) {
+    const auto& relative_uri = serialize_array_uri_to_relative(fragments[i]);
+    entries_builder.set(i, relative_uri);
+  }
+}
+
+void fragments_list_from_capnp(
+    const capnp::ArrayFragmentsList::Reader& array_fragments_list_reader,
+    std::vector<URI>& fragments,
```

Comment:
As per our style guide again, let's make the function return `std::vector<URI>` instead of using an input reference.